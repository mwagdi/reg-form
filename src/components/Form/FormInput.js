import React, { Component } from 'react';

class FormInput extends Component{
    state = {
        input: "",
        conditions: this.props.field.conditions.map(condition => {
            return {
                title: condition.title,
                done: false
            }
        })
    }
    handleChange = e => {
        const { conditions, label } = this.props.field;
        this.setState({ input: e.target.value });
        let cArray = JSON.parse(JSON.stringify(this.state.conditions));
        cArray.forEach((c, i) => {
            c.done = conditions[i].test(e.target.value);
        });
        this.setState({ conditions: [...cArray] }, () => {
            const done = this.state.conditions.reduce((acc, c) => {
                return acc && c.done;
            }, true);
            this.props.onChange(this.state.input, done, label);
        });
    }
    render(){
        const { label, type, name } = this.props.field;
        const id = Math.random().toString(36).substr(2, 16);
        return (
            <div className="regForm__field">
                <label htmlFor={name}>{label}</label>
                <input value={this.state.input} onChange={this.handleChange} name={name} type={type} />
                {this.state.conditions.map((c, i) => <span key={i}>{c.title} - {c.done.toString()}</span>)}
            </div>
        )
    }
}

export default FormInput;