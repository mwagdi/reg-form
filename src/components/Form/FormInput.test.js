import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormInput from './FormInput';

describe('FormInput', () => {
    const field = {
        label: "Text Field",
        type: "text",
        name: "text",
        conditions: [
            {
                title: "Valid text",
                test: str => str.length > 9
            }
        ]
    }
    test('renders without crashing', () => {
        shallow(<FormInput field={field} />);
    });
    test('renders correctly', () => {
        const wrapper = shallow(<FormInput field={field} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('renders div with class regForm__field', () => {
        const wrapper = shallow(<FormInput field={field} />);
        expect(wrapper.exists('.regForm__field')).toEqual(true);
    })
})