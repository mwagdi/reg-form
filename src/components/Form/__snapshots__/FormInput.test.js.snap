// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput renders correctly 1`] = `
<div
  className="regForm__field"
>
  <label
    htmlFor="text"
  >
    Text Field
  </label>
  <input
    name="text"
    onChange={[Function]}
    type="text"
    value=""
  />
  <span
    key="0"
  >
    Valid text
     - 
    false
  </span>
</div>
`;
